src/taipy/_run.py

```diff
@@ -9,22 +9,32 @@
 # an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 # specific language governing permissions and limitations under the License.
 
+import sys
 import typing as t
 
+from flask import Flask
+
 from taipy.gui import Gui
 from taipy.rest import Rest
 from taipy.core import Core
 
+if sys.version_info < (3, 10):
+    from typing_extensions import TypeGuard
+else:
+    from typing import TypeGuard
+
 
-def _run(*apps: t.List[t.Union[Gui, Rest, Core]], **kwargs) -> t.Optional[t.Union[Gui, Rest, Core]]:
+def _run(*apps: t.Union[Gui, Rest, Core], **kwargs) -> t.Optional[t.Union[Gui, Rest, Core, Flask]]:
```

Comment:
This signature was indeed pretty wrong. Thanks.
However, the return type should be `t.Optional[Flask]`.
There would be no use to return... well one of the parameters to `run()`.

Reply:
OK, fixed in next commit.